python/cudf/cudf/core/dtypes.py

```diff
@@ -0,0 +1,66 @@
+import pandas as pd
+from pandas.api.extensions import ExtensionDtype
+
+import cudf
+
+
+class CategoricalDtype(ExtensionDtype):
+    def __init__(self, categories=None, ordered=None):
+        """
+        dtype similar to pd.CategoricalDtype with the categories
+        stored on the GPU.
+        """
+        self._categories = self._init_categories(categories)
+        self.ordered = ordered
+
+    @property
+    def categories(self):
+        if self._categories is None:
+            return None
+        return cudf.core.index.as_index(self._categories)
+
+    @property
+    def type(self):
+        return self._categories.dtype.type
+
+    @property
+    def name(self):
+        return "category"
+
+    @property
+    def str(self):
+        return "|O08"
```

Comment:
Should this number change based on the bitwidth of the indices? i.e. if I have int32 codes this would be 32?

Reply:
I'm not sure what this value should be (https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.dtype.str.html#numpy-dtype-str).

The `CategoricalDtype` knows nothing about the codes so it can't be dependent on that. Pandas always sets this to `|O08`.